chapter2/context-compression/agent.py

```
@@ -563,7 +569,18 @@ def execute_research(self, max_iterations: int = 15) -> Dict[str, Any]:
                         function_name = tool_call['function']['name']
-                        function_args = json.loads(tool_call['function']['arguments'])
+                        raw_args = tool_call['function'].get('arguments') or "{}"
+                        try:
+                            function_args = json.loads(raw_args)
+                            if not isinstance(function_args, dict):
+                                function_args = {}
+                        except json.JSONDecodeError:
+                            # Tolerate bad tool-arg JSON; keep the loop alive.
+                            function_args = {}
+                            logger.warning(
+                                "工具参数不是合法 JSON，已按空对象继续: %r",
```

Ruff (0.16.0)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)